File: packages/renderer-vue/playground/measurements/MeasurementAreaNode.ts
import { defineNode, NodeInterface } from "@baklavajs/core";
import { setType } from "@baklavajs/interface-types";
import { Measurement } from "../models/Measurement";
import { numberType, measurementType } from "../interfaceTypes";

export default defineNode({
    type: "MeasurementAreaNode",
    title: "Area",
    inputs: {
        measurement: () => new NodeInterface<Measurement | undefined>("Measurement", undefined).use(setType, measurementType),
    },
    outputs: {
        area: () => new NodeInterface<number>("Area", 0).use(setType, numberType),
    },
    calculate(inputs) {
        return { area: inputs.measurement?.area };
    },
});

File: BaklavaDependencyEngine/BaklavaDependencyEngine.Core/ExampleNodes/MeasurementAreaNode.cs
using BaklavaDependencyEngine.Core.Models;

namespace BaklavaDependencyEngine.Core.ExampleNodes
{
    public class MeasurementAreaNode : Node
    {
        public MeasurementAreaNode() : base("MeasurementAreaNode", "Measurement Area")
        {
        }

        protected override void InitializeInterfaces()
        {
            AddInput("measurement", new NodeInterface("Measurement", null));
            AddOutput("area", new NodeInterface("Area", 0.0, false));
        }

        public override async Task<Dictionary<string, object>> Calculate(
            Dictionary<string, object> inputs,
            CalculationContext context)
        {
            var measurement = inputs["measurement"] as Measurement;
            var area = measurement?.Area ?? 0.0;

            return await Task.FromResult(new Dictionary<string, object>
            {
                ["area"] = area
            });
        }
    }
}